pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -670,8 +670,8 @@ func (r *ReconcileAWSFederatedAccountAccess) cleanFederatedRoles(reqLogger logr.
 			}
 
 			awsCustomPolicyname := federatedRoleCR.Spec.AWSCustomPolicy.Name + "-" + uidLabel
-			if *policy.PolicyName == awsCustomPolicyname {
-				_, err = awsClient.DeletePolicy(&iam.DeletePolicyInput{PolicyArn: policy.PolicyArn})
+			if *attachedPolicy.PolicyName == awsCustomPolicyname {
```

Comment:
Could still potentially refactor this to something like:
```
err := compareAndDeletePolicy(crPolicyName, policyName, policyArn)
if err != nil {
    return err
}
```
Where `crPolicyName` can have the label applied to it in the func, and all error handling can be managed within the func. This new function can then be unit tested individually.